src/courtroom_widgets.cpp

```diff
@@ -26,9 +26,7 @@ void Courtroom::create_widgets()
   keepalive_timer->start(60000);
 
   chat_tick_timer = new QTimer(this);
-
-  text_delay_timer = new QTimer(this);
-  text_delay_timer->setSingleShot(true);
+  chat_tick_timer->setTimerType(Qt::PreciseTimer); // Prevents drift
```

Comment:
`chat_tick_timer` would benefit from being `singleShot` as it would remove the need to stop. Plus, if the delay becomes 1, then you might end up having more calls than needed, which would lead to slowdowns.

Reply:
Doesn't chat_tick_timer *have* to not be singleShot? We use chat_tick_timer to draw a new character in the messagebox. Having a single shot timer be manually restarted once it expires every time is slow